clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go

```diff
@@ -152,7 +152,17 @@ func (w *waitForControlledPodsRunningMeasurement) start(clients *framework.Multi
 		options.FieldSelector = w.fieldSelector
 		options.LabelSelector = w.labelSelector
 	}
-	listerWatcher := cache.NewFilteredListWatchFromClient(clients.GetClient().CoreV1().RESTClient(), w.kind+"s", w.namespace, optionsModifier)
+	var listerWatcher *cache.ListWatch
+	switch w.kind {
+	case "ReplicationController":
+		listerWatcher = cache.NewFilteredListWatchFromClient(clients.GetClient().CoreV1().RESTClient(), w.kind+"s", w.namespace, optionsModifier)
```

Comment:
This isn't very generic. I would like to avoid introducing switches that you need to expand whenever you want to use a new type...

Reply:
I agree. 
However this PR is just a hot fix for object kinds CL should support. To implement generic approach, I will need definitely more time than this. Thus I thought it will better to create hot fix and then spend time on reimplementation. There is TODO list in #414.

Reply:
I don't think we need this fix that urgently. I would prefer waiting for a proper fix.

Reply:
> I agree.
> However this PR is just a hot fix for object kinds CL should support. To implement generic approach, I will need definitely more time than this. Thus I thought it will better to create hot fix and then spend time on reimplementation. There is TODO list in #414.

I try to implement a dynamic client for informer by package `k8s.io/client-go/dynamic/dynamicinformer`. It works well. 

However, in my private branch, it applies parameter `resource` rather `kind`, since i can't find any way covert type `GroupVersionKind` to type `GroupVersinResource`. Actually, i do not know what difference between `kind` and `resource`.

Reply:
> i can't find any way covert type `GroupVersionKind` to type `GroupVersinResource`

The simplest way: https://github.com/kubernetes/perf-tests/blob/3fb03e234ca8d50a931cbe94800df4265a9df211/clusterloader2/pkg/framework/client/objects.go#L160

Assuming that clusterlaoder uses in all object operations, I think you can also use in this case.